### ISSUE TITLE
Sporadic `Cannot read property 'get' of undefined` error in logs

### ISSUE THREAD

**Issue by novacrazy** [RocketChat/Rocket.Chat#14842]
### Description:

```
=== UnHandledPromiseRejection ===
TypeError: Cannot read property 'get' of undefined
    at PublicRoomType.getNotificationDetails (app/utils/lib/RoomTypeConfig.js:261:29)
    at notifyDesktopUser (app/lib/server/functions/notifications/desktop.js:23:54)
    at Promise.asyncApply (app/lib/server/lib/sendNotificationsOnMessage.js:102:3)
    at /app/bundle/programs/server/npm/node_modules/meteor/promise/node_modules/meteor-promise/fiber_pool.js:43:40
---------------------------------
Errors like this can cause oplog processing errors.
Setting EXIT_UNHANDLEDPROMISEREJECTION will cause the process to exit allowing your service to automatically restart the process
Future node.js versions will automatically exit the process
=================================
```
### Server Setup Information:

- Version of Rocket.Chat Server: 1.2.0-develop (latest as of writing this)
- Operating System: Linux/Ubuntu Server
- Deployment Method: Docker
- Number of Running Instances: 1
- NodeJS Version: 8.15.1
- MongoDB Version: 4.0

### Additional context

It doesn't crash the server, oddly enough, but it's probably not supposed to do that.

**Comment by novacrazy** [comment]
I've had this error hundreds of times today now, and I can't see why `settings` is undefined in that particular spot. Likewise, I don't receive notifications in many channels because of it.

**Comment by novacrazy** [comment]
This should have been solved by #14860
